data-prepper-core/src/main/java/org/opensearch/dataprepper/pipeline/Pipeline.java

```diff
@@ -211,7 +211,21 @@ public void execute() {
                 final SourceCoordinator sourceCoordinator = sourceCoordinatorFactory.provideSourceCoordinator(partionProgressModelClass, name);
                 ((UsesSourceCoordination) source).setSourceCoordinator(sourceCoordinator);
             }
-            source.start(buffer);
+
+            sinkExecutorService.submit(() -> {
+                while (!isReady()) {
```

Comment:
What happens if `/shutdown` is called while we are waiting for sinks to be ready? Can we call stop on a source that is not started yet? Have we tested this?